language/module.md

```diff
@@ -42,9 +42,15 @@ already explains what a module is and what it can contain.
 | `/examples/`          | Example Puppet snippets that explain how to use the module. They can be used within acceptance tests |
 | `/facts.d/`           | [External facts](https://www.puppet.com/docs/puppet/latest/external_facts.html) that are synced via [pluginsync](https://www.puppet.com/docs/puppet/latest/plugins_in_modules.html) |
 | `/lib/facter/`        | MAY contain custom facts |
-| `/lib/puppet/type/`   | Custom types |
-| `/lib/puppet/provider/` | Custom provider for one or multiple types |
+| `/lib/puppet/type/`   | Custom Resource types |
+| `/lib/puppet/provider/` | Custom provider for one or multiple Resource types |
```

Comment:
Technically, modules can put whatever they want in the `/lib` directory and it will be packaged in the module and pluginsynced to agents. So from an allowlist perspective, listing individual `/lib/puppet/*` directories is misleading. But from a "what are the various extension points" perspective, it's informative.

Reply:
> Technically, modules can put whatever they want in the /lib directory
is that something that we should allow here? Do you have anything in mind that isn't already covered by the different `/lib/puppet/` subdirs?

Reply:
> is that something that we should allow here?

I think we should allow anything under `/lib` since that's how puppet behaves today. For example, there was some early confusion about `/lib/puppetx` vs `/lib/puppet_x`. And given `/lib/augeas` there may be some others?

In the future, I think it would be good to document which extension points are allowed in the `/lib` directory and add a deprecation warning to if we detect a dir/file that isn't allowed during pluginsync. Then in a major release, only pluginsync files that are in the allow list.

Reply:
> add a deprecation warning to if we detect a dir/file that isn't allowed during pluginsync

Do you think it's realistic to introduce the warnings in Puppet 8 or maybe with Puppet 9.0?